app/src/main/java/org/simple/clinic/setup/SetupActivity.kt

```diff
@@ -126,27 +112,19 @@ class SetupActivity : AppCompatActivity(), UiActions {
   }
 
   override fun showOnboardingScreen() {
-    screenRouter.popAndPush(OnboardingScreenKey(), RouterDirection.FORWARD)
+    // If onboarding screen is already being shown don't navigate again, it would cause
+    // duplicate destinations
+    if (navController.currentDestination?.id != R.id.onboardingScreen) {
+      navController.navigate(R.id.action_placeholderScreen_to_onboardingScreen)
+    }
```

Comment:
Do we need to do this check all the time? If so, we can move this check into `ScreenNavigator`, no?

Reply:
Yeah, the reason why I didn't move it inside navigator is we may need to launch a destination even though it's already a current destination. So I didn't want to change the expected behaviour of the library to avoid issues when we are migrating to the fragment-based navigation in future.

Reply:
How does the default fragment based implementation handle this case?

Reply:
We don't have to do it for every navigate call. This particular call was being made everytime when we are restoring state, so to avoid duplicates I placed this check.

Reply:
I had to use the same check with when using fragments based navigation as well.

Reply:
> We don't have to do it for every navigate call. This particular call was being made everytime when we are restoring state, so to avoid duplicates I placed this check.

I see. Then I think the actual problem here is the way we are restoring state in `SetupActivity` since it is making a navigation call after state restoration, correct? If this is the case, then we can fix that problem later.

I think `Flow` internally checks if the same key is being pushed and handles this case itself, so moving to nav controller exposes a problem in our UI setup which we did not know about.